drivers/at24cxxx/at24cxxx.c

```diff
@@ -0,0 +1,366 @@
+    xtimer_usleep(AT24CXXX_MAX_WRITE_CYCLE * US_PER_MS);
+    return AT24CXXX_OK;
+}
+
+static int _write(const at24cxxx_t *dev, uint32_t pos, const uint8_t *data,
+                  size_t len)
+{
+    size_t wr = 0;
+    size_t chunk_len;
+    int check = 0;
+
+    while ((chunk_len = len - (size_t)wr)) {
+        uint8_t buffer[AT24CXXX_ADDR_LEN];
+        _position_to_address_words(buffer, pos);
+
+        check = i2c_write_bytes(dev->params.i2c, dev->params.dev_addr, buffer,
+                                sizeof(buffer), I2C_NOSTOP);
+        DEBUG("write_bytes check: %d\n", check);
+        if (check < 0) {
+            break;
+        }
+
+        const uint8_t *chunk_data = data + wr;
+        size_t max_chunk_len =
+            (AT24CXXX_PAGE_SIZE - (pos % AT24CXXX_PAGE_SIZE));
+        chunk_len = chunk_len < max_chunk_len ? chunk_len : max_chunk_len;
+        chunk_len = chunk_len <
+                    AT24CXXX_WRITE_CHUNK_SIZE ? chunk_len :
+                    AT24CXXX_WRITE_CHUNK_SIZE;
+
+        check = i2c_write_bytes(dev->params.i2c, dev->params.dev_addr,
+                                chunk_data, chunk_len, I2C_NOSTART);
+        DEBUG("write_bytes check: %d\n", check);
+        if (check < 0) {
+            break;
+        }
+        wr += chunk_len;
+        pos += chunk_len;
+        xtimer_usleep(AT24CXXX_MAX_WRITE_CYCLE * 1000);
+    }
+    return wr > 0 ? (int)wr : check;
+}
+
+static int _set(const at24cxxx_t *dev, uint32_t pos, uint8_t val, size_t len)
+{
+    size_t wr = 0;
+    size_t chunk_len;
+    uint8_t chunk[AT24CXXX_WRITE_CHUNK_SIZE] = { val };
+    int check = 0;
+
+    while ((chunk_len = len - (size_t)wr)) {
+        uint8_t buffer[AT24CXXX_ADDR_LEN];
+        _position_to_address_words(buffer, pos);
+
+        check = i2c_write_bytes(dev->params.i2c, dev->params.dev_addr, buffer,
+                                sizeof(buffer), I2C_NOSTOP);
+        DEBUG("write_bytes check: %d\n", check);
+        if (check < 0) {
+            break;
+        }
+
+        size_t max_chunk_len =
+            (AT24CXXX_PAGE_SIZE - (pos % AT24CXXX_PAGE_SIZE));
+        chunk_len = chunk_len < max_chunk_len ? chunk_len : max_chunk_len;
+        chunk_len = chunk_len <
+                    AT24CXXX_WRITE_CHUNK_SIZE ? chunk_len :
+                    AT24CXXX_WRITE_CHUNK_SIZE;
+
+        check = i2c_write_bytes(dev->params.i2c, dev->params.dev_addr, chunk,
+                                chunk_len, I2C_NOSTART);
+        DEBUG("write_bytes check: %d\n", check);
+        if (check < 0) {
+            break;
+        }
+        wr += chunk_len;
+        pos += chunk_len;
+        xtimer_usleep(AT24CXXX_MAX_WRITE_CYCLE * 1000);
+    }
+    return wr > 0 ? (int)wr : check;
+}
+
+const at24cxxx_params_t *at24cxxx_get_default_dev_params(void)
+{
+    return &at24cxxx_params[0];
+}
+
+size_t at24cxxx_eeprom_size(at24cxxx_dev_type_t dev_type)
+{
+    return (
+        (dev_type == AT24CXXX_DEV_TYPE_AT24C256) ? AT24C256_EEPROM_SIZE
```

Comment:
This would be much more readable as a switch - case.

Reply:
Ok, I will rewrite that.